File: software/examples/ruby/example_button_callbacks.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_lcd_20x4'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = '6mJ' # Change to your UID

ipcon = IPConnection.new # Create IP connection
lcd = BrickletLCD20x4.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register button status callbacks
lcd.register_callback(BrickletLCD20x4::CALLBACK_BUTTON_PRESSED) do |i|
  puts "Pressed: #{i}"
end
lcd.register_callback(BrickletLCD20x4::CALLBACK_BUTTON_RELEASED) do |i|
  puts "Released: #{i}"
end

puts 'Press key to exit'
$stdin.gets

File: software/examples/python/example_button_callbacks.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_lcd_20x4 import LCD20x4

# Callback functions for button status
def cb_pressed(i):
    print('Pressed: ' + str(i))

def cb_released(i):
    print('Released: ' + str(i))

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    lcd = LCD20x4(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Register button status callbacks to cb_pressed and cb_released
    lcd.register_callback(lcd.CALLBACK_BUTTON_PRESSED, cb_pressed)
    lcd.register_callback(lcd.CALLBACK_BUTTON_RELEASED, cb_released)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()
